apps/omg_api/test/state/transaction_test.exs

```diff
@@ -204,4 +204,59 @@ defmodule OMG.API.State.TransactionTest do
 
     assert {:error, :malformed_metadata} = Transaction.Signed.decode(ExRLP.encode([sigs, inputs, outputs, ""]))
   end
+
+  @tag fixtures: [:utxos, :alice]
+  test "Decoding transaction with gaps in inputs returns error",
+       %{utxos: [input1, input2 | _], alice: %{addr: alice_addr}} do
+    assert {:error, :inputs_contain_gaps} ==
```

Comment:
I'm not 100% certain, but I'd suggest testing on a more "public API level", i.e. `TestHelper.create_signed |> Transaction.Recovered.recover_from` pair (actually `TestHelper.create_encoded |> decode |> ...`, but the former is better for now, as I'll likely refactor `Transaction.Recovered.recover_from` to take bytes as input while #529)

The behavior that we really care about is what happens when processing

Reply:
To make this testable as you suggested @pdobacz I've introduced quite [unpleasant hack](https://github.com/omisego/elixir-omg/blob/b261e5ba4c5e462a9e15bfc1d7f0bef1c8004541/apps/omg_api/lib/state/transaction/recovered.ex#L42). 
I want to make sure that `no gaps check` is really low that no incorrect transaction can sneak in.

Reply:
ah I hear you now. Yeah let's do this till #529 lands (and remember to fix differently, if it doesn't :) )